src/coreclr/jit/codegenwasm.cpp

```diff
@@ -895,9 +895,22 @@ void CodeGen::genCompareFloat(GenTreeOp* treeNode)
 {
     assert(treeNode->OperIsCmpCompare());
 
+    bool invertSense = false;
+
     if ((treeNode->gtFlags & GTF_RELOP_NAN_UN) != 0)
     {
-        NYI_WASM("genCompareFloat: unordered compares");
+        // Wasm float comparisons other than "fne" return false for NaNs.
+        // Our unordered float compares need to return true for NaNs.
+        //
+        // So we can re-express say GT_GE (UN) as !GT_LT
+        //
+        if (!treeNode->OperIs(GT_NE))
+        {
+            treeNode->ChangeOper(GenTree::ReverseRelop(treeNode->OperGet()));
+            invertSense = true;
+        }
+
+        treeNode->gtFlags &= ~GTF_RELOP_NAN_UN;
```

Comment:
It would be cleaner to just have a local variable for the inverted oper instead of mutating the node like this.

Reply:
Yes

Reply:
Done